### PR TITLE
Use FluentAssertions to validate Result and Result<T>

### PR DESCRIPTION
**Pull request by Corniel**
[FluentAssertions](https://fluentassertions.com/) is the best way of writing assertions in tests I've seen. It has been highly adopted, and is so in line with what Qowaiv wants to deliver (code that is Domain Driven) that creating extensions to make asserting on `Result` and `Result<T>` is not more than logical.

The API created is simple (at least, that is the attempt), and tries to help with writing meaningful tests:

``` C#
Result.For(13).IsValid().WithoutMessages().Value.Should().Be(13); 
```
Where the chain after `.Value` is regular FluentAssertions.

Other options for `.IsValid()` are:

``` C#
Result.For(13).IsValid().WithMessage(ValidationMessage.Warn("Are you sure?")).Value.Should().Be(13);
````
and

``` C#
Result.For(13).IsValid().WithMessages(/* ... */).Value.Should().Be(13);
````

Obviously, you can also check for invalidness:

``` C#
Result.For(13).IsInValid(); 
```
In that case you do not have the option of `.WithoutMessages()`, obviously, as `Result` is defined to be invalid bases on containing at least one error message. Furthermore, it does not exposes the `.Value` as that is not accessible for an invalid result.

All (except for the `.Value`) is also for the non-generic `Result`.

By doing this, I made all other tooling obsolete, as I think their is no need/value in supporting both.

### CODE REVIEW

File: src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using Qowaiv.Validation.Abstractions;
+
+namespace FluentAssertions.Qowaiv.Validation
+{
+    /// <summary>Contains a number of methods to assert the (in)validness of a <see cref="Result"/>.</summary>
+    public sealed class ResultAssertions : ResultAssertionsBase<Result>
+    {
+        /// <summary>Creates a new instance of the <see cref="ResultActExtensions"/> class.</summary>
```
Comment (mars888) [inline src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs:8]: Is this comment correct?

File: src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertionsBase.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using System.Linq;
+using System.Text;
+
+namespace FluentAssertions.Qowaiv.Validation
+{
+    /// <summary>Contains a number of methods to assert the (in)validness of a <see cref="Result"/>.</summary>
+    public abstract class ResultAssertionsBase<TSubject>
+        where TSubject : Result
+    {
+        /// <summary>Creates a new instance of the <see cref="ResultActExtensions"/> class.</summary>
```
Comment (mars888) [inline src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertionsBase.cs:12]: Is this comment correct?

File: src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions_T.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using Qowaiv.Validation.Abstractions;
+
+namespace FluentAssertions.Qowaiv.Validation
+{
+    /// <summary>Contains a number of methods to assert the (in)validness of a <see cref="Result{TModel}"/>.</summary>
+    public sealed class ResultAssertions<TModel> : ResultAssertionsBase<Result<TModel>>
+    {
+        /// <summary>Creates a new instance of the <see cref="ResultActExtensions"/> class.</summary>
```
Comment (mars888) [inline src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions_T.cs:8]: Is this comment correct?

File: src/Qowaiv.Validation.TestTools/FluentAssertions/ResultValidnessMessageAssertions_T.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿using Qowaiv.Validation.Abstractions;
+
+namespace FluentAssertions.Qowaiv.Validation
+{
+    /// <summary>Contains a number of methods to assert the messages or value  of a valid <see cref="Result{TModel}"/>.</summary>
```
Comment (mars888) [inline src/Qowaiv.Validation.TestTools/FluentAssertions/ResultValidnessMessageAssertions_T.cs:5]: Extra space in line.

**Review by mars888 — APPROVED** [review]

File: src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs (hunk 1 above)
Comment (Corniel) [reply src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs:8]: I think so, do you have a better/other idea?

File: src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertionsBase.cs (hunk 2 above)
Comment (Corniel) [reply src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertionsBase.cs:12]: I think so, do you have a better/other idea?

File: src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions_T.cs (hunk 3 above)
Comment (Corniel) [reply src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions_T.cs:8]: I think so, do you have a better/other idea?

File: src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs (hunk 1 above)
Comment (mars888) [reply src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs:8]: I was mostly wondering, here and in the other comments, if the class listed is actually the correct one (probably should have been clearer on this one 😅).

Comment (Corniel) [reply src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs:8]: I think so. Maybe, the naming could be better, and I'm open to suggestions, but the paths disallow `.WithoutMessages()` for invalid, and `.Value` is only accessible for `.Should().BeValid()` on `Result<T>`.

Comment (mars888) [reply src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs:8]: I'm not sure if we are talking about the same thing here. The comment mentions creating an instance of `ResultActExtensions`, but the class here is `ResultAssertions`. But could be I'm missing something as it's a bit harder to trace stuff in the online diff view.

Comment (Corniel) [reply src/Qowaiv.Validation.TestTools/FluentAssertions/ResultAssertions.cs:8]: Ill check if all of them match. :) I thought you're referring to meaning of the text, not the type mismatches.

**Pull request merged**